packages/parameters/src/BaseProvider.ts

```diff
@@ -35,8 +40,9 @@ abstract class BaseProvider implements BaseProviderInterface {
    * this should be an acceptable tradeoff.
    * 
    * @param {string} name - Parameter name
-   * @param {GetOptionsInterface} options - Options to configure maximum age, trasformation, AWS SDK options, or force fetch
+   * @param {GetOptionsInterface|SSMGetOptionsInterface} options - Options to configure maximum age, trasformation, AWS SDK options, or force fetch
    */
+  public async get(name: string, options?: SSMGetOptionsInterface): Promise<undefined | string | Record<string, unknown>>;
   public async get(name: string, options?: GetOptionsInterface): Promise<undefined | string | Record<string, unknown>> {
```

Comment:
Yes, it's a function overload which as far as I have been reading, it's purely a TypeScript construct. Since JavaScript proper doesn't really have function overload you are allowed to specify only one implementation (not sure if this answers your question).

I tried union types initially but I ended up with overloading both because of the inputs but also because of the outputs. In this specific case `SSMProvider.get` never returns an `Uint8Array` while other providers do (i.e. `AppConfigProvider`).

If I use the union type the type-checker complains like this:
![image](https://user-images.githubusercontent.com/7353869/210098293-0e96baea-3bee-4cef-814d-08a9be4e15e2.png)

I'm very open to alternatives though. I have to admit that this is very close to the limits of my TS knowledge so if there's a better / more elegant way that allows users to get the right types when using the library I'm open to consider it.